source/wp-content/plugins/wporg-internal-notes/src/sidebar.js

```diff
@@ -27,9 +29,34 @@ const SidebarIcon = () => {
 };
 
 export const NotesSidebar = () => {
-	const initialNotes = useSelect( ( select ) => {
-		return select( notesStore ).getNotes();
+	const { initialNotes, afterDate, isSaving } = useSelect( ( select ) => {
+		return {
+			initialNotes: select( notesStore ).getNotes(),
+			afterDate: select( notesStore ).getLatestNoteDate(),
+			isSaving: select( 'core/editor' ).isSavingPost(),
+		};
 	} );
+	const { prependNotes, clearIsCreated } = useDispatch( notesStore );
+	const prevIsSaving = usePrevious( isSaving );
+	const [ isLoading, setIsLoading ] = useState( false );
+
+	// Update the notes list when new notes are created, even if the sidebar is closed.
+	useEffect( () => {
+		// Gutenberg doesn't have a specific way to check whether the post was just saved, so we store the
+		// previous value of isSaving and when isSaving goes from true -> false, we prepend the notes.
+		const postWasJustSaved = prevIsSaving && ! isSaving;
+		if ( postWasJustSaved ) {
+			setIsLoading( true );
+			prependNotes( afterDate ).then( ( { notes } ) => {
```

Comment:
I didn't notice anything in testing, but could using the date cause any issues with accidentally skipping notes?

Reply:
I guess if somehow two log notes were created with exactly the same timestamp, but only one of them was initially loaded into the sidebar, the other one would get skipped... but that seems like an extreme edge case. Plus, if you refreshed the screen, the skipped log would then appear, so very little harm would be done.

Reply:
Fair enough, I agree with the harmless edge-case judgement 🙂